### PR TITLE
currency conversion api: unittests, frontend added. 

### PR DESCRIPTION
**Pull request by ihsangkcl**
currency conversion api: unittests, frontend added. Solved some minor errors in frontend files. Added some mistakenly deleted parts for currency convert api.

### CODE REVIEW

**Comment by veyisTurgut** [comment]
Zero conflicts, consistent with the structure, nice tests. 
Thanks İhsan.
Approved.
I'll wait for one more review and then we can merge directly.

**Review by veyisTurgut — APPROVED** [review]
Commented my review above

**Comment by gokayyildiz** [comment]
zero conflicts, can be directly merged 
I think earlier we merge, better we continue

**Pull request merged**
